### ISSUE TITLE
How to debug the sdmgr_head.py

### ISSUE THREAD

**Issue by lyc728** [open-mmlab/mmocr#403]
I want to debug and print the edge information
        for gnn_layer in self.gnn_layers:
            nodes, cat_nodes = gnn_layer(nodes, embed_edges, node_nums)

        node_cls, edge_cls = self.node_cls(nodes), self.edge_cls(cat_nodes)
        print('edge_cls()')
But I can't get the result
![fc8497f1aef99aad9c1186bdd4b636b5](https://user-images.githubusercontent.com/88084847/127488135-ec228306-b995-4b98-845e-3a09db5d50fa.jpg)

**Comment by gaotongxiao** [comment]
For your first code snippet, you are printing `'edge_cls()'` which is a string, instead of a variable. Use `print(edge_clas)` instead.

As long as you add this print statement and start the training process, you should be able to see something like 
```
tensor([[ 0.0350, -0.0239],
        [ 0.0455, -0.0313],
        [ 0.0475, -0.0272],
        ...,
        [ 0.0509, -0.0368],
        [ 0.0449, -0.0365],
        [ 0.0467, -0.0367]], device='cuda:0', grad_fn=<AddmmBackward>)
``` 
If you want the program to print all the values out, try
```
import numpy
import sys
numpy.set_printoptions(threshold=sys.maxsize)
print(embed_edges.detach().cpu().numpy())
```

Meanwhile, since we have to put the highest priority on the development tasks, our apologies for not being able to answer your questions in time. It's a better idea to raise your questions in instant messenger groups, where there may be more people available to answer your questions.

**Comment by lyc728** [comment]
> For your first code snippet, you are printing `'edge_cls()'` which is a string, instead of a variable. Use `print(edge_clas)` instead.
> 
> As long as you add this print statement and start the training process, you should be able to see something like
> 
> ```
> tensor([[ 0.0350, -0.0239],
>         [ 0.0455, -0.0313],
>         [ 0.0475, -0.0272],
>         ...,
>         [ 0.0509, -0.0368],
>         [ 0.0449, -0.0365],
>         [ 0.0467, -0.0367]], device='cuda:0', grad_fn=<AddmmBackward>)
> ```
> 
> If you want the program to print all the values out, try
> 
> ```
> import numpy
> import sys
> numpy.set_printoptions(threshold=sys.maxsize)
> print(embed_edges.detach().cpu().numpy())
> ```
> 
> Meanwhile, since we have to put the highest priority on the development tasks, our apologies for not being able to answer your questions in time. It's a better idea to raise your questions in instant messenger groups, where there may be more people available to answer your questions.

I added it as you suggested, but it didn't show either
        if x is not None:
            nodes = self.fusion([x, nodes])

        all_edges = torch.cat(
            [rel.view(-1, rel.size(-1)) for rel in relations])
        embed_edges = self.edge_embed(all_edges.float())
        embed_edges = F.normalize(embed_edges)

        numpy.set_printoptions(threshold=sys.maxsize)
        print(embed_edges.detach().cpu().numpy())

<img width="705" alt="微信图片_20210730085937" src="https://user-images.githubusercontent.com/88084847/127584109-4e4f000d-f8e4-474a-874e-6601a31d1384.png">
